lib/vmdb/settings.rb

```diff
@@ -88,13 +88,30 @@ def self.build_template
     private_class_method :build_template
 
     def self.build(resource)
-      build_template.tap do |settings|
-        settings.add_source!(DatabaseSource.new(resource))
+      build_without_local(resource).tap do |settings|
         local_sources.each { |s| settings.add_source!(s) } if resource.try(:is_local?)
       end
     end
     private_class_method :build
 
+    def self.parent_settings_without_local(resource)
+      build_template.tap do |settings|
+        DatabaseSource.parent_sources_for(resource).each do |db_source|
+          settings.add_source!(db_source)
+        end
+      end
+    end
+    private_class_method :parent_settings_without_local
+
+    def self.build_without_local(resource)
+      build_template.tap do |settings|
+        DatabaseSource.sources_for(resource).each do |db_source|
+          settings.add_source!(db_source)
+        end
+      end
+    end
+    private_class_method :build_without_local
```

Comment:
yes, this code as well as corresponding  DatabaseSource.source_for/.paren_source_for could be oprimized, will do it in separate PR